app/marketing/views.py

```diff
@@ -407,15 +412,30 @@ def account_settings(request):
 
         if request.POST.get('disconnect', False):
             profile.github_access_token = ''
+            profile = record_form_submission(request, profile, 'account-disconnect')
             profile.email = ''
             profile.save()
             messages.success(request, _('Your account has been disconnected from Github'))
             logout_redirect = redirect(reverse('logout') + '?next=/')
             return logout_redirect
         if request.POST.get('delete', False):
+            # remove profile
             profile.hide_profile = True
+            profile = record_form_submission(request, profile, 'account-delete')
             profile.email = ''
             profile.save()
+
+            # remove email
+            try:
+                client = MailChimp(mc_user=settings.MAILCHIMP_USER, mc_api=settings.MAILCHIMP_API_KEY)
+                result = client.search_members.get(query=es.email)
+                subscriber_hash = result['exact_matches']['members'][0]['id']
+                client.lists.members.delete(
+                    list_id=settings.MAILCHIMP_LIST_ID,
+                    subscriber_hash=subscriber_hash,
+                    )
+            except Exception as e:
```

Comment:
Which exception(s) are we expecting here? IndexError and KeyError per line 432 and probably a `MailChimp` connection error?

Reply:
any exception.  the point is that any exception that happens shouldn't cause the page to 500